genesyscloud/resource_exporter/resource_name_sanitizer.go

```diff
@@ -99,7 +113,45 @@ func (sod *sanitizerOptimized) Sanitize(idMetaMap ResourceIDMetaMap) {
 }
 
 // SanitizeResourceName sanitizes a single resource name
-func (sod *sanitizerOptimized) SanitizeResourceName(inputName string) string {
+func (sod *sanitizerNameOptimized) SanitizeResourceName(inputName string) string {
+	name := unsafeNameChars.ReplaceAllStringFunc(inputName, escapeRune)
+
+	if unsafeNameStartingChars.MatchString(string(rune(name[0]))) {
+		// Terraform does not allow names to begin with a number. Prefix with an underscore instead
+		name = "_" + name
+	}
+
+	return name
+}
+
+// Sanitize sanitizes all resource name using the time optimized algorithm
+func (sod *sanitizerTimeOptimized) Sanitize(idMetaMap ResourceIDMetaMap) {
+	sanitizedNames := make(map[string]int, len(idMetaMap))
+
+	for _, meta := range idMetaMap {
+		sanitizedName := sod.SanitizeResourceName(meta.Name)
+
+		if sanitizedName != meta.Name {
+			if count, exists := sanitizedNames[sanitizedName]; exists {
+				// We've seen this sanitized name before
+				sanitizedNames[sanitizedName] = count + 1
+
+				// Append a hash to ensure uniqueness
+				h := sha256.New()
+				h.Write([]byte(meta.Name))
+				hash := hex.EncodeToString(h.Sum(nil)[:10]) // Use first 10 characters of hash
```

Comment:
Amazon Q suggested to use this hashing algorithm due to better efficiency and reduced collisions. Considering this would change the output of the names, I wrapped this whole functionality in a GENESYS_SANITIZER_TIME_OPTIMIZED env variable feature flag.